src/realm_dart_sync.cpp

```diff
@@ -139,56 +139,46 @@ RLM_API void realm_dart_sync_on_subscription_state_changed_callback(realm_userda
     });
 }
 
-bool wait_dart_before_continue(realm::util::UniqueFunction<void(realm::util::UniqueFunction<void(bool)>*)>* userCallback)
+bool invoke_dart_and_await_result(realm::util::UniqueFunction<void(realm::util::UniqueFunction<void(bool)>*)>* userCallback)
 {
     std::condition_variable condition;
     std::mutex mutex;
-    bool complete = false;
     bool success = false;
+    bool completed = false;
 
     realm::util::UniqueFunction<void(bool)> unlockFunc = [&](bool result) {
         std::unique_lock<std::mutex> lock(mutex);
-        complete = true;
         success = result;
+        completed = true;
         lock.unlock();
         condition.notify_one();
     };
     (*userCallback)(&unlockFunc);
 
     std::unique_lock<std::mutex> lock(mutex);
```

Comment:
```suggestion
    std::unique_lock lock(mutex);
```
let compiler infer

Reply:
that's not something this PR is changing but I prefer your suggestion. will commit